### PR TITLE
TodoApp. Updated Kotlin to 1.7.10, Compose to 1.2.0, and other libs

### PR DESCRIPTION
**Pull request by arkivanov**
Updated versions, migrated the code.

Kotlin: 1.7.10
Compose: 1.2.0
Gradle: 7.4.2
AGP: 7.2.0
Decompose: 1.0.0-alpha-06
MVIKotlin: 3.0.0

Also removed `iosWorkaroundSupportArm64Simulator`.

### CODE REVIEW

File: examples/todoapp/gradle.properties [hunk 1]
```diff
@@ -23,6 +23,5 @@ org.gradle.parallel=true
 org.gradle.caching=true
 kotlin.native.disableCompilerDaemon=true
 
-#todo remove -Pkotlin.version=1.6.20 from Xcode project, when stable version on Compose with Koltin 1.6.20 or later released
-kotlin.version=1.6.10
-compose.version=1.1.0
\ No newline at end of file
+kotlin.version=1.7.0
+compose.version=1.2.0-alpha01-dev745
```
Comment (igordmn) [inline examples/todoapp/gradle.properties:27]: Thanks! But let's keep this PR until we release a stable version of 1.2.0. The general rule for `/examples` and `/tutorials` is to use a stable version of Compose.

Comment (igordmn) [reply examples/todoapp/gradle.properties:27]: 1.2.0 is out. We can merge `master` to this PR, solving conflicts (new versions are already there), and the merge this PR to `master`

**Review by igordmn — APPROVED** [review]

File: examples/todoapp/gradle.properties (hunk 1 above)
Comment (arkivanov) [reply examples/todoapp/gradle.properties:27]: Sure, I will update the PR soon.

Comment (arkivanov) [reply examples/todoapp/gradle.properties]: I have pushed updates, also updated the PR description.

File: examples/todoapp/buildSrc/buildSrc/src/main/kotlin/Deps.kt [hunk 2]
```diff
@@ -29,7 +34,7 @@ object Deps {
     object Android {
         object Tools {
             object Build {
-                const val gradlePlugin = "com.android.tools.build:gradle:7.0.4"
+                const val gradlePlugin = "com.android.tools.build:gradle:7.2.0"
```
[the remark this answers is not in the document]
Comment (arkivanov) [reply examples/todoapp/buildSrc/buildSrc/src/main/kotlin/Deps.kt:42]: I have tried AGP `7.1.3` and hit some [build errors](https://github.com/JetBrains/compose-jb/issues/2340#issuecomment-1277735753) with Lint. Meantime, I applied the workaround that I described in that issue. Let me know if this works for you.

[the remark this answers is not in the document]
Comment (igordmn) [reply examples/todoapp/buildSrc/buildSrc/src/main/kotlin/Deps.kt:42]: Let's keep this workaround.

Could you also add to `gradlew.properties`:
```
// TODO remove workaround after upgrading to AGP 7.3
// After updating Compose Multiplatform version, update corresponding Jetpack Compose versions in Android module (search "Workaround for https://github.com/JetBrains/compose-jb/issues/2340")
```

and solve conflicts with `master` branch?

[the remark this answers is not in the document]
Comment (arkivanov) [reply examples/todoapp/buildSrc/buildSrc/src/main/kotlin/Deps.kt:42]: Sure! Just one question - why `gradle.properties`? Maybe right there in `Deps.kt`, where the AGP version is defined?

[the remark this answers is not in the document]
Comment (igordmn) [reply examples/todoapp/buildSrc/buildSrc/src/main/kotlin/Deps.kt:42]: We need to update them when we update the main Compose version.

We keep this version in `gradle.properties` to easly override it on CI.

[the remark this answers is not in the document]
Comment (arkivanov) [reply examples/todoapp/buildSrc/buildSrc/src/main/kotlin/Deps.kt:42]: Done, please review.

**Comment by igordmn** [comment]
Thank again. All is fine )

**Pull request merged**